hkdf/src/hkdf.rs

```diff
@@ -42,10 +45,11 @@ impl<D> Hkdf<D>
     }
 
     /// The RFC5869 HKDF-Expand operation
-    pub fn expand(&self, info: &[u8], length: usize) -> Vec<u8> {
+    pub fn expand<T>(&self, info: &[u8], mut out: T) where T: AsMut<[u8]> {
```

Comment:
Do we need a generic `T: AsMut<[u8]>` here? I think a simple `&mut [u8]` should suffice, and it will be more coherent with other crates API. Also it will be nice to make `expand` panic-free function by returning `Result`.